esmvalcore/preprocessor/_area.py

```diff
@@ -295,3 +302,86 @@ def extract_named_regions(cube, regions):
     constraints = iris.Constraint(region=lambda r: r in regions)
     cube = cube.extract(constraint=constraints)
     return cube
+
+
+def _clip_geometries(cube, geometries):
+    """Clip cube to only include the geometries and padding."""
+    lon_coord = cube.coord(axis='X')
+    lat_coord = cube.coord(axis='Y')
+    if lon_coord.ndim == 1 and lat_coord.ndim == 1:
+        (
+            start_longitude,
+            start_latitude,
+            end_longitude,
+            end_latitude,
+        ) = geometries.bounds
+        # add a padding of one cell around the clipped cube
+        lon_bound = lon_coord.core_bounds()[0]
+        lon_step = lon_bound[1] - lon_bound[0]
+        start_longitude -= lon_step
+        end_longitude += lon_step
+        lat_bound = lat_coord.core_bounds()[0]
+        lat_step = lat_bound[1] - lat_bound[0]
+        start_latitude -= lat_step
+        end_latitude += lat_step
+        cube = extract_region(cube, start_longitude, end_longitude,
+                              start_latitude, end_latitude)
+
+    return cube
+
+
+def extract_shape(cube, shapefile, method='contains', clip=True):
+    """Extract a region defined by a shapefile.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+       input cube.
+    shapefile: str
+        A shapefile defining the region(s) to extract.
+    method: str
+        Select all points contained by the shape ('contains') or
+        select a single representative point ('representative').
+    clip: bool
+        Clip the resulting cube ('true') or not ('false').
```

Comment:
can we think of any used case when `clip` could be False?

Reply:
Yes, if you prefer keeping your data global

Reply:
wut? if you extract a region you obviously don't want your data global :grin:

Reply:
as in, all is masked anyway so your region of interest is the only bit you care for and multimodel operations will be uselelss anyway since you need to unmask, which is redundant

Reply:
That's why the default is to clip, but if you're selecting a region that is most of the globe, I think it makes sense to keep the entire dataset.

Reply:
We're unable to clip irregular grids too, so if you have datasets of both kinds that you want to use with your diagnostic, you probably do not want to clip either.